pandas/core/arrays/base.py

```diff
@@ -679,13 +710,105 @@ def convert_values(param):
 
             return res
 
-        op_name = _get_op_name(op, True)
+        op_name = ops._get_op_name(op, True)
         return set_function_name(_binop, op_name, cls)
 
 
-class ExtensionScalarArithmeticMixin(ExtensionScalarOpsMixin):
+def ExtensionArithmeticOpsMixin(base=ExtensionDefaultOpsMixin):
```

Comment:
Why is it needed to make this a factory function? 
The alternative is make this a class and then the user needs to implement a `_create_method` in its ExtensionArray ?

Reply:
@jorisvandenbossche I don't see an easy way of sharing the implementation of `_create_method()` that is in `ExtensionScalarOpsMixin` with the two mixin classes `ExtensionScalarArithmeticMixin` and `ExtensionScalarComparisonMixin`.  I think it's because `_create_method()` is a class method and when `ExtensionArithmeticOpsMixin` is set up, it needs to know the base class that contains the class method `_create_method()`.

The alternative is to do it how I did it before - i.e., just make `ExtensionScalarOpsMixin` a base class.